src/main/java/org/broadinstitute/hellbender/utils/collections/IntervalTree.java

```diff
@@ -0,0 +1,999 @@
+package org.broadinstitute.hellbender.utils.collections;
+
+import org.broadinstitute.hellbender.utils.CollatingInterval;
+
+import java.util.ConcurrentModificationException;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+/**
```

Comment:
It's depressing that we need a 3rd copy of this code.  Can we abstract it slightly so we can have 1 instead of 3?